plugins/collections/update.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.collection.base, 'id');
+
+module.exports = () => ({
+    method: 'PUT',
+    path: '/collections/{id}',
+    config: {
+        description: 'Update a collection',
+        notes: 'Update a specific collection',
+        tags: ['api', 'collection'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const { id } = request.params;
+            const { collectionFactory, userFactory } = request.server.app;
+            const { username } = request.auth.credentials;
+
+            // get the collection and user
+            return Promise.all([
+                collectionFactory.get({ id }),
+                userFactory.get({ username })
+            ])
+            .then(([oldCollection, user]) => {
+                if (!oldCollection) {
+                    throw boom.notFound(`Collection ${id} does not exist`);
+                }
+
+                const oldCollectionUserId = oldCollection.toJson().userId;
```

Comment:
@FenrirUnbound never mind figured it out. It was because my GET route was previously mutating the testCollection model object in memory before but it works now after I deep cloned it in the GET route.